apps-rendering/src/components/liveBlocks.tsx

```diff
@@ -0,0 +1,97 @@
+// ----- Imports ----- //
+
+import { css } from '@emotion/react';
+import { FirstPublished } from '@guardian/common-rendering/src/components/FirstPublished';
+import { LastUpdated } from '@guardian/common-rendering/src/components/LastUpdated';
+import LiveBlockContainer from '@guardian/common-rendering/src/components/liveBlockContainer';
+import { headline, space } from '@guardian/source-foundations';
+import { OptionKind, partition } from '@guardian/types';
+import type { DeadBlog, LiveBlog } from 'item';
+import type { FC } from 'react';
+import { renderAll } from 'renderer';
+
+// ----- Component ----- //
+interface LiveBlocksProps {
+	item: DeadBlog | LiveBlog;
+}
+
+interface BlockTitleProps {
+	title: string;
+}
+
+const BlockTitle: FC<BlockTitleProps> = ({ title }) => {
+	return (
+		<h2
+			css={css`
+				${headline.xxsmall({ fontWeight: 'bold' })}
+				margin-bottom: ${space[2]}px;
+			`}
+		>
+			{title}
+		</h2>
+	);
+};
+
+const LiveBlocks: FC<LiveBlocksProps> = ({ item }) => {
+	return (
+		<div>
+			{/* Accordion? */}
+			{/* Pagination? */}
+			{item.blocks.map((block) => {
+				// TODO: get page number
+				const blockLink = `${1}#block-${block.id}`;
+
+				return (
+					<LiveBlockContainer
+						key={block.id}
+						id={block.id}
+						borderColour={'black'}
+					>
+						<header
+							css={css`
+								padding-right: ${space[3]}px;
+								display: flex;
+								flex-direction: column;
+							`}
+						>
+							{block.firstPublished.kind === OptionKind.Some && (
+								<FirstPublished
+									firstPublished={Number(
+										block.firstPublished.value,
+									)}
+									blockLink={blockLink}
+								/>
+							)}
```

Comment:
There's a helper function in `lib.ts` that you may find useful here (_may_, it depends what you find more readable):

```suggestion
							{maybeRender(block.firstPublished, firstPublished =>
								<FirstPublished
									firstPublished={Number(firstPublished)}
									blockLink={blockLink}
								/>
							)}
```

Couple more thoughts on this:

1. Do you think it's valid to render a block if `firstPublished` is missing? Should we actually consider invalidating the whole block if we don't know when it was published? If so, this is probably something we'd want to do a step earlier during parsing 🤔 
2. Would it make sense for the `FirstPublished` component to take a `Date` instead? I've noticed that internally it's converting the `number` into a `Date` anyway. Also it's a component named after a datetime so semantically I think it makes sense?

Reply:
I really like `maybeRender` - thanks for highlighting this :)

1) Maybe I've missed your point, are we suggesting not showing a blog post to a reader if we don't know the firstPublished date? My initial feeling is: show as much content as possible, as long as the content itself can be successfully rendered. However, maybe there are legal requirements around displaying the published dateTime that I'm not aware of (or some other consideration entirely).

I think in dcr we perform equivalent logic and so we've, perhaps lazily, replicated equivalent functionality here in the hope that we'd be able to share both the Header and its children (https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/components/LiveBlock.tsx#L74)

2) About the props for `FirstPublished` - totally agree, when we were pairing we found it frustrating to convert the prop to `Number`, only for it to be converted back to a `Date` inside the component. 

For this skeleton PR I'm tempted to leave as it is - we have a specific checklist item to review shared components and I've included an extra note to review `FirstPublished` props as part of this (https://trello.com/c/3PIfKTMB). Given we have a task to review the existing shared components and their props do you think it's acceptable to leave as-is for now? I think it might be 'cleaner' to raise a separate PR to tidy up these props as it would impact dcr too. Let me know what you think about this approach :)

Reply:
> are we suggesting not showing a blog post to a reader if we don't know the firstPublished date?

That's what I think we should determine, yes. Perhaps we should have a chat about this?

> Given we have a task to review the existing shared components and their props do you think it's acceptable to leave as-is for now?

Yup sounds reasonable.

Reply:
We discussed this problem and I think we agreed:
- it would be simpler to only parse valid blocks (e.g. in liveBlock.ts) - rejecting any blocks that do not have a firstPublished or lastModified date. Then in the layout file our rendering logic could be simplified because we'd only have valid blocks (this means we could remove some of the maybeRender functions).
- I have added this task as a new checklist item into the AR liveblocks card so we can address this under a separate PR (https://trello.com/c/3PIfKTMB)

If the above sounds reasonable @JamieB-gu are you happy to approve?